### DISCUSSION TITLE
How do I add Vaultwarden behind proxy?

### DISCUSSION THREAD

**Discussion by batijo** [tteck/Proxmox#783] [Help Needed]
So I installed Vaultwarden and Nginx Proxy Manager, what are my next steps? 

> ⚠️ Vaultwarden needs to be behind a proxy (Nginx Proxy Manager) to obtain HTTPS and to allow clients to connect.

I'm new to all this self hosting thing and have no idea how to configure SSL on Vaultwarden. I hope it isn't wrong place to ask this question.

**Comment by tteck** [comment]
Setup a domain name (free at duckdns.org)

Then study this 

https://github.com/hassio-addons/addon-nginx-proxy-manager/raw/main/images/screenshot.gif

Instead of Home Assistant, use the Vaultwarden IP & port

**Reply by batijo** [reply]
Thanks for reply!

As I understand using domain name requires you to open ports on your router am I right?

Is there are a way to add SSL to your local domain (vaultwarden.local)?
I've tried it and it doesn't work. What I did is pointed my vaultwarden.local domain to proxy manager and then just added a proxy host to forward it to Vaultwarden IP and that worked but once I tried adding certificate it gave me internal error.

**Reply by dimatx** [reply]
Using a domain with SSL doesn't require opening of ports as long as you use DNS verification (an option you'll see in Nginx when creating a certificate.)
